kontrol-service/engine/flow/dev_flow.go

```diff
@@ -206,7 +204,8 @@ func applyPatch(
 	}
 
 	modifiedTargetService := DeepCopyService(targetService)
-	modifiedTargetService.DeploymentSpec = deploymentSpec
+	// TODO: find a better way to update deploymentSpec, this assumes there is only container in the pod
+	modifiedTargetService.DeploymentSpec.Template.Spec.Containers[0].Image = newImage
```

Comment:
the target service deployment spec could have been updated when executing the external plugins before these lines, for instance, there could be some env vars overwriting that we need to keep in the deployment spec.
The new image is the only value we need to maintain from the flow_patch.

Reply:
maybe we should think on a `MergeSpec(a Spec, b Spec) Spec` function instead. Then we decide how to merge it inside the function (like `a` always take precedence)